space_packet_parser/xtce/definitions.py

```diff
@@ -434,7 +482,7 @@ def packet_generator(
             show_progress: bool = False,
             buffer_read_size_bytes: Optional[int] = None,
             skip_header_bytes: int = 0
-    ) -> Iterator[Union[packets.CCSDSPacket, UnrecognizedPacketTypeError]]:
+    ) -> Iterator[Union[packets.Packet, UnrecognizedPacketTypeError]]:
```

Comment:
So if we had a non-CCSDS packet, would we need a separate generator method (and associated generator function)? This method _sounds_ agnostic to CCSDS packets (`packet_generator()`) but internally it only works for CCSDS packets. 

I'm not sure how to handle that better. i.e. I'm not sure how to decide whether to call the CCSDS generator or the non-CCSDS generator. Do we want the user telling us whether they are using CCSDS packets or not when initializing the packet definition object? That would make it awkward to support mixed definitions where some are CCSDS and others are not.

Reply:
Yes, this is definitely CCSDS specific, because that is where we started from. We could do something like "bring your own generator" as an input argument. But, this sort of gets at a definition probably shouldn't have the generator on it, the generator should be separate and then each yielded item can be parsed by the `definition.parse_bytes()` method. Or we would grow a separate generator for each yielded possibility.

Reply:
Per your Slack message, I think you suggested we make this generator part of the definition but I'm not sure I fully understand how that would work.